submissions/VBystrov/memory-pair-game/js/index.js

```diff
@@ -0,0 +1,166 @@
+import { backCover } from '../js/back.js';
+const amountCards = 25;
+const rows = 4;
+const columns = 6;
+const cardsInGame = (rows * columns) / 2;
+const showCardsTime = 1500;
+const delayCardShow = 300;
+
+class Game {
+  constructor(cardsContainer, newGameButton) {
+    this.cardsContainer = cardsContainer;
+    this.newGameButton = newGameButton;
+    this.allCardCodes = [];
+    this.pairedCards = [];
+    this.fieldLocked = false;
+
+    this.addAllCardCodes();
+
+    this.newGameButton.addEventListener('click', () => {
+      this.start();
+    });
+
+    this.cardsContainer.addEventListener('click', (e) => {
+      const card = e.target.closest('.card');
+
+      if (this.pairedCards.includes(card) || !card || this.fieldLocked) {
+        return;
+      }
+      this.reverseCards(card);
+
+      if (!(this.pairedCards.length & 1)) {
+        this.pairedCards.push(card);
+        return;
+      } else {
+        if (
+          this.isCardsPair(card, this.pairedCards[this.pairedCards.length - 1])
+        ) {
+          this.pairedCards.push(card);
+          this.pairedCards.slice(-2).forEach((newPairCard) => {
+            newPairCard.classList.add('pair');
+          });
+
+          if (this.pairedCards.length >= cardsInGame * 2) {
+            this.newGameButton.disabled = false;
+          }
+        } else {
+          this.reverseNotPairedCards(card);
+        }
+      }
+    });
+  }
+
+  start() {
+    this.newGameButton.disabled = true;
+    this.pickCards();
+    this.shuffleCards();
+    this.placeCards();
+    this.revealAllCards();
+  }
+
+  addAllCardCodes() {
+    for (let i = 0; i < amountCards; i++) {
+      this.allCardCodes.push(i.toString(10).padStart(3, '0'));
```

Comment:
This may be an overkill. Using `.toString()` with radix argument would make sense to me, if you would convert base 10 numbers to some other base. But you can leave it as is if you like.

Reply:
Maybe it is. Removed.